node/connection_base.js

```diff
@@ -113,6 +114,13 @@ TChannelConnectionBase.prototype.onTimeoutCheck = function onTimeoutCheck() {
     }
     if (self.lastTimeoutTime) {
         self.timedOutEvent.emit(self);
+    } else if (!self.remoteName &&
+        self.timers.now() - self.startTime > self.channel.requestDefaults.timeout) {
+        self.timedOutEvent.emit(self, errors.TimeoutError({
```

Comment:
Who should listen to this error and close the connection? 
Currently, connection listens to the timeout event and closes itself.  Should I make a similar handler for time out error?